backend/main-service/src/main/java/kkakka/mainservice/coupon/domain/Discount.java

```diff
@@ -0,0 +1,80 @@
+package kkakka.mainservice.coupon.domain;
+
+import java.time.LocalDateTime;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import kkakka.mainservice.category.domain.Category;
+import kkakka.mainservice.product.domain.Product;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Getter
+@Table(name = "Discount")
+@AllArgsConstructor
```

Comment:
정적 팩토리 메서드를 사용한다면 
```suggestion
@AllArgsConstructor(access = AccessLevel.PRIVATE)
```
이렇게 외부에서 생성자를 직접 호출하는걸 막아주면 좋을 것 같아요!

Reply:
넵 좋은 리뷰 감사합니다 😁